src/ConnectedMode/Migration/MSBuildFileProvider.cs

```diff
@@ -31,18 +38,111 @@ namespace SonarLint.VisualStudio.ConnectedMode.Migration
     [PartCreationPolicy(CreationPolicy.NonShared)]
     internal class MSBuildFileProvider : IFileProvider
     {
+        private readonly IServiceProvider serviceProvider;
         private readonly ILogger logger;
+        private readonly IThreadHandling threadHandling;
+        private readonly IFileSystem fileSystem;
+
+        internal static readonly string[] FileSearchPatterns = new string[] {
+            "*.ruleset",
+            "*.props",
+            "*.targets",
+            "*.csproj",
+            "*.vbproj"
```

Comment:
`Directories.GetFiles` only matches one pattern at a time, so we need to call it multiple times and aggregate the results, dropping out any duplicates.

I don't have test for de-duplication - I'll add one.